### PR TITLE
ISSUE-4374: Apply simd selection to StringColumn

### LINKED ISSUE #4374: Feature: Simd selection in BooleanColumn && StringColumn
**Summary**

Description for this feature.

Continue #4264 

Let's apply simd selection into BooleanColumn && StringColumn.

### PR DESCRIPTION
**Pull request by LiuYuHui**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog
- Performance Improvement

## Related Issues

Fixes #4374 

## Test Plan

Unit Tests

### CODE REVIEW

File: common/datavalues/src/columns/string/mod.rs [hunk 1]
```diff
@@ -189,15 +191,39 @@ impl Column for StringColumn {
...
         let values = self.values();
-        for (i, v) in filter.values().iter().enumerate() {
-            if v {
-                let start = self.offsets[i] as usize;
-                let end = self.offsets[i + 1] as usize;
-                builder.append_value(&values[start..end]);
-            }
-        }
+        let (slice, _, length) = filter.values().as_slice();
+        let mut mask_chunks = BitChunksExact::<u64>::new(slice, length);
```
[the remark this answers is not in the document]
Comment (sundy-li) [reply common/datavalues/src/columns/string/mod.rs:199]: See: L202 in : 
https://github.com/datafuselabs/databend/blob/43d5d5e01ed3f7fff11e5fc0f6e82c315832eeda/common/datavalues/src/columns/primitive/mod.rs

[the remark this answers is not in the document]
Comment (sundy-li) [reply common/datavalues/src/columns/string/mod.rs:221]: Boolean::filter also encounter this problem in your previous pr.

So I suggest:

Add a test `let col = column.slice(n, size);  col.filter()` to cover this case  (0 < n < 8)

[the remark this answers is not in the document]
Comment (LiuYuHui) [reply common/datavalues/src/columns/string/mod.rs:221]: I have a question about the filter in `BooleanColumn`, should I consider the case that  `value_offset !=0` in `self.values`?
``` rust
let (value_slice, value_offset, value_length) = self.values().as_slice();
let (slice, offset, length) = filter.values().as_slice();
```
I understand `offset` in `filter.values` should be considered.

[the remark this answers is not in the document]
Comment (sundy-li) [reply common/datavalues/src/columns/string/mod.rs:221]: Yes, we should consider about that.

The `value_offset` will be not zero if we  slice the column using `let col = column.slice(n, len);`

[the remark this answers is not in the document]
Comment (LiuYuHui) [reply common/datavalues/src/columns/string/mod.rs:221]: Hi @sundy-li , when I working on the case `offset !=0`, I realize that we don't need care `offset !=0`, even `offset !=0`, we still can use `BitChunksExact::<u64>::new(slice, length);`, right?
 In my unit test, ignore`offset!=0`still works, please correct me if my understanding is wrong.

[the remark this answers is not in the document]
Comment (sundy-li) [reply common/datavalues/src/columns/string/mod.rs:221]: ` ignoreoffset!=0still work`

It happens to work, but it's not right if you understand the right way.

[the remark this answers is not in the document]
Comment (sundy-li) [reply common/datavalues/src/columns/string/mod.rs:221]: The offset makes sense.

File: common/datavalues/tests/it/columns/string.rs [hunk 2]
```diff
@@ -49,3 +49,61 @@ fn test_string_column() {
...
+                    "hello".as_bytes()
+                }
+            })),
+        },
+        Test {
+            filter: BooleanColumn::from_iterator((0..N).map(|_| false)),
+            expect: NewColumn::new_from_iter(empty_case.iter()),
+        },
+        Test {
+            filter: BooleanColumn::from_iterator((0..N).map(|i| i % 3 == 0)),
```
Comment (sundy-li) [inline common/datavalues/tests/it/columns/string.rs:87]: If you boolean is more random, you can cover the offset case I think.

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
@mergify update

**Pull request merged**